File: android/app/src/main/kotlin/com/typography/animations/entrance/Flip3D_X.kt
package com.typography.animations.entrance

import com.typography.AnimatedTextState

fun evaluateFlip3DX(t: Float): AnimatedTextState {
    val rotationX = -90.0f * (1.0f - t)
    val scale = 0.6f + 0.4f * t
    return AnimatedTextState(
        opacity = t.coerceIn(0f, 1f),
        rotationX = rotationX,
        scale = scale
    )
}

File: lib/utils/animations/entrance/flip3d_x.dart
import '../base/animated_text_state.dart';

AnimatedTextState evaluateFlip3DX(double t) {
  final rotationX = -90.0 * (1.0 - t);
  final scale = 0.6 + 0.4 * t;
  return AnimatedTextState(
    opacity: t.clamp(0.0, 1.0),
    rotationX: rotationX,
    scale: scale,
  );
}
